### PR TITLE
Add agent config for tolerations, affinity, resources on cluster

### PR DESCRIPTION
**Pull request by manno**
Changes are tracked via the, hashed, status fields and will redeploy the agent.

This also adds more code comments to the custom resources.

refers to #1392

### CODE REVIEW

File: pkg/agent/manifest.go [hunk 1]
```diff
@@ -229,6 +231,16 @@ func agentDeployment(namespace string, agentScope string, opts ManifestOptions,
...
 	dep.Spec.Template.Spec.Tolerations = append(dep.Spec.Template.Spec.Tolerations, opts.AgentTolerations...)
 
+	// overwrite affinity if present on cluster
+	if opts.AgentAffinity != nil {
+		dep.Spec.Template.Spec.Affinity = opts.AgentAffinity
+	}
+
+	// set resources if present on cluster
+	if opts.AgentResources != nil {
+		dep.Spec.Template.Spec.Containers[0].Resources = *opts.AgentResources
```
Comment (raulcabello) [inline pkg/agent/manifest.go:241]: shouldn't we change Resources in all containers? Imagine we have a pod with multiple containers, I think it should be applied to all containers.

File: pkg/controllers/manageagent/manageagent_test.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
+package manageagent
+
+import (
+	"testing"
+
+	// counterfeiter -o internal/fakes github.com/rancher/wrangler/pkg/generated/controllers/core/v1.NamespaceController
```
Comment (raulcabello) [inline pkg/controllers/manageagent/manageagent_test.go:6]: can we remove the counterfeiter comment as we are using gomock?

**Review by raulcabello — CHANGES_REQUESTED** [review]
I left a comment regarding applying `Resources` to all containers instead of just the first one.

Unit tests are great!

File: pkg/agent/manifest.go (hunk 1 above)
Comment (manno) [reply pkg/agent/manifest.go:241]: The agent deployment is only one container. This is in line with the env vars below, they're also added just to the first container.

File: pkg/controllers/manageagent/manageagent_test.go (hunk 2 above)
Comment (manno) [reply pkg/controllers/manageagent/manageagent_test.go]: I turned that into a `//go:generate` comment. `go generate ./pkg/controllers/manageagent` will re-create the mock.

File: pkg/agent/manifest.go (hunk 1 above)
Comment (raulcabello) [reply pkg/agent/manifest.go:241]: what if someone is using something like Istio that injects a sidecar container?

**Review by raulcabello — APPROVED** [review]

**Pull request merged**